nextcord/abc.py

```diff
@@ -684,6 +684,11 @@ def permissions_for(self, obj: Union[Member, Role], /) -> Permissions:
             denied = Permissions.all_channel()
             base.value &= ~denied.value
 
+        # if you are timed out then you lose all permissions except view_channel and read_message_history
+        if obj.communication_disabled_until is not None:
```

Comment:
Has this been tested with obj being a `Role` object? I feel this might cause a problem.

Reply:
This code will never be run on a `Role` object, the comment I added was removed earlier, but this code handles the `Member` case only. `Role` is handled earlier and always returns earlier.

Reply:
Tested and seems fine. Role returns early, so does not reach this code.